### PR TITLE
NP-1609: Keep latest publication when listing doi-requests

### PR DESCRIPTION
**Pull request by axthosarouris**
When listing the DoiRequests for an institution, we currently present all version of a published publication. With this fix we list only the latest version of a published publication

### CODE REVIEW

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 1]
```diff
@@ -98,9 +100,24 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
...
+            .parallel()
+            .collect(Collectors.groupingBy(Publication::getIdentifier))
+            .values()
+            .stream()
+            .flatMap(this::mostRecentPublication)
+            .collect(Collectors.toList());
+    }
+
+    private Stream<Publication> mostRecentPublication(List<Publication> publicationList) {
+        return publicationList.stream().max(Comparator.comparing(Publication::getModifiedDate)).stream();
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:118]: newlines between method call?

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 2]
```diff
@@ -219,7 +238,7 @@ private void putItem(Publication publication) {
     }
 
     private Publication fetchPublicationById(UUID publicationId) throws NotFoundException {
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:240]: Call this publicationIdentifier  (identifier for UUIDs and IDs for IRIs)

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 3]
```diff
@@ -254,11 +273,10 @@ private Item publicationToItem(Publication publication) {
         return Optional.empty();
     }
 
-    private QuerySpec queryForLastestPublicationWithId(UUID publicationId) {
-        QuerySpec query = new QuerySpec()
+    private QuerySpec queryForLatestPublicationWithId(UUID publicationId) {
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:276]: Call this publicationIdentifier (identifier for UUIDs and IDs for IRIs)

**Review by norrs — CHANGES_REQUESTED** [review]
Will this also fix https://unit.atlassian.net/browse/NP-1608 ?

**Comment by axthosarouris** [comment]
> Will this also fix https://unit.atlassian.net/browse/NP-1608 ?

No, NP-1608 is related to search and it also of another nature (it shows non-published publications while here we have duplicates of published publications.)

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java (hunk 1 above)
Comment (axthosarouris) [reply doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java]: fixed

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 4]
```diff
@@ -113,7 +116,7 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
 
     @Override
     public Optional<Publication> fetchDoiRequestByPublicationId(UUID publicationId) throws NotFoundException {
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:118]: nitpick: Method name as well.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 5]
```diff
@@ -132,12 +135,29 @@ public void createDoiRequest(CreateDoiRequest createDoiRequest, String username)
     @Override
     public void updateDoiRequest(UUID publicationID, DoiRequestStatus requestedStatusChange, String requestedByUsername)
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:136]: nitpick: parameter

**Review by norrs — APPROVED** [review]

**Review by norrs — APPROVED** [review]

**Pull request merged**